book/continuous_integration.md

```diff
@@ -0,0 +1,171 @@
+# Automating Code Quality Assurance
+
+## Motivation
+
+There are various tasks which can be automated to increase the quality of code and make development easier and less tedious. Automating the running of unit tests is especially important to ensuring trust in your pipeline or package, by ensuring that all unit tests pass before every merge.
+
+
+(automate-tests)=
+### Automate tests to reduce risk of errors
+
+Tests should be run whenever you make changes to your project.
+This ensures that changes do not break the existing, intended functionality of your code.
+However, it is easy to forget to run your tests at regular intervals.
+
+"Surely this could be automated too?"
+
+Absolutely! Automatic testing, amongst other quality assurance measures, can be triggered when changes are made to your remote version control repository.
+These tools can be used to ensure that all changes to a project are tested.
+Additionally, it allows others, who are reviewing your code, to see the results of your tests.
+
+
+(continuous-integration)=
+## Commit small changes often
+
+Committing small changes regularly is often referred to as Continuous Integration (CI).
```

Comment:
We might need to make this definition a bit clearer.

Reply:
We've specifically not tried to define CI as other sources are often conflicting and the point here is to encourage quality assurance of code and agile working.